### PR TITLE
Deprecate support for docker.elastic.co/elasticsearch/elasticsearch-oss

### PR DESCRIPTION
**Pull request by dadoonet**
Since version 7.11.0, there is no more `-oss` artefact anymore.
This simplifies the code as we don't need to check anymore what is available and what is not.

### CODE REVIEW

**Review by kiview — DISMISSED** [review]
That's very nice and simplifies the module as well as UX for Testcontainers and Elasticsearch users. Do you see any potential for this breaking for users that have pull-through caches configured and will therefore continue using old `-oss` suffixed images?

**Comment by dadoonet** [comment]
Is there anything else needed from my side to merge this?

File: modules/elasticsearch/src/main/java/org/testcontainers/elasticsearch/ElasticsearchContainer.java [hunk 1]
```diff
@@ -32,13 +32,17 @@ public class ElasticsearchContainer extends GenericContainer<ElasticsearchContai
...
+
+    @Deprecated
     private static final DockerImageName DEFAULT_OSS_IMAGE_NAME = DockerImageName.parse("docker.elastic.co/elasticsearch/elasticsearch-oss");
 
     /**
      * Elasticsearch Default version
      */
     @Deprecated
-    protected static final String DEFAULT_TAG = "7.9.2";
+    protected static final String DEFAULT_TAG = "7.15.0";
```
Comment (eddumelendez) [inline modules/elasticsearch/src/main/java/org/testcontainers/elasticsearch/ElasticsearchContainer.java:43]:
```suggestion
    protected static final String DEFAULT_TAG = "7.9.2";
```

File: modules/elasticsearch/src/main/java/org/testcontainers/elasticsearch/ElasticsearchContainer.java [hunk 2]
```diff
@@ -51,15 +55,15 @@ public ElasticsearchContainer() {
 
     /**
      * Create an Elasticsearch Container by passing the full docker image name
-     * @param dockerImageName Full docker image name as a {@link String}, like: docker.elastic.co/elasticsearch/elasticsearch:7.9.2
+     * @param dockerImageName Full docker image name as a {@link String}, like: docker.elastic.co/elasticsearch/elasticsearch:7.15.0
```
Comment (eddumelendez) [inline modules/elasticsearch/src/main/java/org/testcontainers/elasticsearch/ElasticsearchContainer.java:58]:
```suggestion
     * @param dockerImageName Full docker image name as a {@link String}, like: docker.elastic.co/elasticsearch/elasticsearch:7.9.2
```

File: modules/elasticsearch/src/main/java/org/testcontainers/elasticsearch/ElasticsearchContainer.java [hunk 3]
```diff
@@ -51,15 +55,15 @@ public ElasticsearchContainer() {
...
      */
     public ElasticsearchContainer(String dockerImageName) {
         this(DockerImageName.parse(dockerImageName));
     }
 
     /**
      * Create an Elasticsearch Container by passing the full docker image name
-     * @param dockerImageName Full docker image name as a {@link DockerImageName}, like: DockerImageName.parse("docker.elastic.co/elasticsearch/elasticsearch:7.9.2")
+     * @param dockerImageName Full docker image name as a {@link DockerImageName}, like: DockerImageName.parse("docker.elastic.co/elasticsearch/elasticsearch:7.15.0")
```
Comment (eddumelendez) [inline modules/elasticsearch/src/main/java/org/testcontainers/elasticsearch/ElasticsearchContainer.java:66]:
```suggestion
     * @param dockerImageName Full docker image name as a {@link DockerImageName}, like: DockerImageName.parse("docker.elastic.co/elasticsearch/elasticsearch:7.9.2")
```

File: modules/elasticsearch/src/test/java/org/testcontainers/elasticsearch/ElasticsearchContainerTest.java [hunk 4]
```diff
@@ -30,7 +30,7 @@ public class ElasticsearchContainerTest {
     /**
      * Elasticsearch version which should be used for the Tests
      */
-    private static final String ELASTICSEARCH_VERSION = "7.9.2";
+    private static final String ELASTICSEARCH_VERSION = "7.15.0";
```
Comment (eddumelendez) [inline modules/elasticsearch/src/test/java/org/testcontainers/elasticsearch/ElasticsearchContainerTest.java:33]:
```suggestion
    private static final String ELASTICSEARCH_VERSION = "7.9.2";
```

File: modules/elasticsearch/src/test/java/org/testcontainers/elasticsearch/ElasticsearchContainerTest.java [hunk 5]
```diff
@@ -211,6 +203,32 @@ public void restClientSecuredClusterHealth() throws IOException {
...
+                .build();
+
+            Response response = client.performRequest(new Request("GET", "/_cluster/health"));
+            // }}
+            assertThat(response.getStatusLine().getStatusCode(), is(200));
+            assertThat(EntityUtils.toString(response.getEntity()), containsString("cluster_name"));
+        }
+        // }
+    }
+
```
Comment (eddumelendez) [inline modules/elasticsearch/src/test/java/org/testcontainers/elasticsearch/ElasticsearchContainerTest.java:231]: There are already tests for version 8

```suggestion
```

**Comment by dadoonet** [comment]
I'd suggest to use 7.17.15 as the latest 7.x version: https://www.elastic.co/downloads/past-releases/elasticsearch-7-17-15

**Comment by eddumelendez** [comment]
@dadoonet thanks for the suggestion, but default constructor is deprecated and if by any chance there is someone still using it there will be a surprise by using a different image. Users must declare the image to use.

**Comment by dadoonet** [comment]
Ha! Indeed. I misread the PR. You actually reverted the change to 7.15. So all good on my end. Thansk for moving forward this PR ;)

**Comment by eddumelendez** [comment]
Thanks for your contribution, @dadoonet ! Let's discuss via slack or using GH discussions the next steps proposed in the first comment. I think some of them were already addressed.

**Pull request merged**